internal/common/autogen/handle_operations.go

```diff
@@ -118,6 +119,39 @@ func HandleRead(ctx context.Context, req HandleReadReq) {
 	req.Resp.Diagnostics.Append(req.Resp.State.Set(ctx, req.State)...)
 }
 
+// HandleDataSourceReadReq contains the request parameters for a data source read operation.
+type HandleDataSourceReadReq struct {
+	Resp       *datasource.ReadResponse
+	Client     *config.MongoDBClient
+	Config     any // The configuration model (TFDSModel)
+	CallParams *config.APICallParams
+}
+
+// HandleDataSourceRead handles the read operation for a data source.
+func HandleDataSourceRead(ctx context.Context, req HandleDataSourceReadReq) {
```

Comment:
we could have `HandleDataSourceRead` and `HandleRead` both call a same private function that has parametrized through a function how not found scenario is handled. This will help avoid any deviations in the future.